src/dune_rules/ctypes_stanza.mli

```diff
@@ -41,20 +41,40 @@ module Type_description : sig
 end
 
 module Function_description : sig
-  type t =
-    { concurrency : Concurrency_policy.t
-    ; errno_policy : Errno_policy.t
-    ; functor_ : Module_name.t
-    ; instance : Module_name.t
-    }
+  module Info : sig
+    type t =
+      { concurrency : Concurrency_policy.t
+      ; errno_policy : Errno_policy.t
+      }
+  end
+
+  module Key : sig
+    type t =
+      { external_library_name : External_lib_name.t
```

Comment:
How come we need this new field? It's already present in the record that includes the info map.

Reply:
Intermediate file names are built from exactly these pieces of info (lib name, functor name, instanciated name), so this is the triplet that needs to be unique. So it's used as a key. This lib name was always passed next to the other fields.

This refactoring might be a bit too involved. If it's too confusing I can remove this commit and leave it in the previous architecture (where a map is built to check uniqueness with a nice error message, but thrown away).

Reply:
> This refactoring might be a bit too involved. If it's too confusing I can remove this commit and leave it in the previous architecture (where a map is built to check uniqueness with a nice error message, but thrown away).

That seems cleaner. We aren't doing any key lookups in this map so tossing it away sounds like the right thing to do. Additionally, this `Key` module would remain private.